ember-core/shared/src/main/scala/org/http4s/ember/core/h2/PseudoHeaders.scala

```diff
@@ -59,6 +59,12 @@ private[h2] object PseudoHeaders {
   }
 
   def headersToRequestNoBody(hI: NonEmptyList[(String, String)]): Option[Request[fs2.Pure]] = {
+    def toRawHeader(k: String, v: String): Header.ToRaw =
+      Header.ToRaw.rawToRaw(Header.Raw(org.typelevel.ci.CIString(k), v))
+
+    def isLowerCase(str: String): Boolean =
```

Comment:
This one I think may create more allocated objects than it saves us.

Reply:
Are you referring to the `String.forall` boxing the Char characters?

This is the implementation in the library: 

```Scala
  def forall(@deprecatedName("f", "2.13.3") p: Char => Boolean): Boolean = {
    var i = 0
    val len = s.length
    while(i < len) {
      if(!p(s.charAt(i))) return false
      i += 1
    }
    true
  }
```

Reply:
The `StringOps` is just a syntax-extension "decorator" class, and it does not inherit from any collection-generic trait that may require a `forall` with an `AnyRef` on it... so it should be unboxed... but only the decompiler can tell, to be sure.